libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -642,23 +690,46 @@ public boolean matches(@NotNull Attributes attrs) {
 
     @Override
     public void onMediaUploadProgress(final String localMediaId, final float progress) {
-        if(!isAdded()) {
+        if(!isAdded() || content == null || !mAztecReady) {
             return;
         }
-        final MediaType mediaType = mUploadingMedia.get(localMediaId);
-        if (mediaType != null) {
-            AztecText.AttributePredicate localMediaIdPredicate = ImagePredicate.getLocalMediaIdPredicate(localMediaId);
-            content.setOverlayLevel(localMediaIdPredicate, 1, (int)(progress * 10000));
-            content.refreshText();
+
+        // first obtain the latest maximum
+        float maxProgressForLocalMediaId = mUploadingMediaProgressMax.get(localMediaId);
```

Comment:
This can throw an NPE if the `localMediaId` doesn't exist in `mUploadingMediaProgressMax` (`HashMap.get()` will return a null `Float`, which will crash when unboxing into a `float`).